sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -152,6 +152,7 @@ public Mono<Void> logRequest(ClientLogger logger, HttpRequestLoggingContext logg
                     .append(request.getHttpMethod())
                     .append(" ")
                     .append(getRedactedUrl(request.getUrl(), allowedQueryParameterNames))
+                    .append(" ") // ensure the gap between url and upcoming contents if line separators are removed
```

Comment:
The interaction of logging framework and the logging itself is quite confusing for me.

I kind of assume logging framework should not touch the logging itself. If someone feel multiple line could be a problem, they should fix the logging, not the `ClientLogger` that affect all the logging as well as customer's logging as well (but what is done is done...).

Another joke, dev from another team asks, why in each retry, our payload increase 100M? We find that the "Try count: 1" + "body size" create something like "Try count: 199823443-byte body", "Try count: 299823443-byte body".

However changing this level of logging could be risky. One cannot be sure whether customer has been set-up a logging parse policy to parse and collection these to their internal analysis framework.

Reply:
That is very good feedback @weidongxu-microsoft on `ClientLogger` trying to fix newline attacks instead of the underlying logging configurations. It was done for cases where this isn't being protected but it's a valid point that an SDK shouldn't be handling this and that the underlying implementation should.

@JonathanGiles @srnagar do you have any thoughts on this? If we want to discuss further we can create a separate issue for it.

Reply:
There is also another path where logs coming from the SDKs themselves, or at least their inputs, can be marked as safe as we're the owners of the content

Reply:
Let's discuss separately.